2D_Generator/src/IsingHamiltonian.h

```diff
@@ -0,0 +1,80 @@
+#pragma ONCE
+#include <iostream>
+#include <stdio.h>
+#include <vector>
+#include <chrono>
+#include "slepcsys.h"
+#include <slepceps.h>
+#include "petscsys.h"
+#include <fstream>
+
+
+#include "debug_flags.hpp"
+
+
+#ifndef PW
+/* Macro for checking PETSC calls */
+#define PW(call) petsc_wrap(call, __LINE__)
+
+#define dprint(text) std::cout << text << ": " << __LINE__ << std::endl;
+
+#define vprint(...) if(flags.verbose_mode) {  PetscPrintf(MPI_COMM_WORLD, __VA_ARGS__ ); PetscSynchronizedFlush(PETSC_COMM_WORLD, PETSC_STDOUT );}
+#define nprint(text) if(flags.verbose_mode) { {int rank; MPI_Comm_rank(MPI_COMM_WORLD, &rank); std::cout << "Node " << rank <<": " << text << std::endl;} }
```

Comment:
I can see the 440 influence haha

Reply:
> The 2D generator can be configured to solve a 1D system as well, assuming the qubits are arranged in a ring, however, this feature still needs to be verified. Use the flag `--1d` as denoted in the README file.

I think it makes sense to use the old, slow Eigen-based solver, and see if the two outputs match.

Reply:
Or just use python, and just get rid of the Eigen-based one